quipucords/scanner/network/runner/roles/user_data/tasks/main.yml

```diff
@@ -0,0 +1,57 @@
+---
+
+- name: internal_host_started_processing_role
+  set_fact:
+    internal_host_started_processing_role: "user_data"
+
+# Facts to determine number of users
+- name: gather etc/passwd data fact
+  raw: tail -5 /etc/passwd
+  register: internal_user_info_cmd
+  ignore_errors: yes
+  become: yes
+  when: 'user_has_sudo'
+
+- name: set internal_user_info_cmd fact
+  set_fact:
+    internal_user_info: "{{ internal_user_info_cmd }}"
+  ignore_errors: yes
+
+- name: initialize user_info_data
+  set_fact:
+    user_info: "QPC_FORCE_POST_PROCESS"
+  ignore_errors: yes
+
+- name: gather user_login_history data fact
+  raw: last -25
+  register: internal_user_login_history_cmd
+  ignore_errors: yes
+  become: yes
+  when: 'user_has_sudo'
+
+- name: set internal_user_login_history_cmd fact
+  set_fact:
+    internal_user_login_history: "{{ internal_user_login_history_cmd }}"
+  ignore_errors: yes
+
+- name: initialize user_login_history
+  set_fact:
+    user_login_history: "QPC_FORCE_POST_PROCESS"
+  ignore_errors: yes
+
+- name: gather user_delete_history data fact
+  raw: cat /var/log/secure 2>/dev/null | grep userdel
```

Comment:
So, I think this command is missing some of the data. So, the issue with only looking at `/var/log/secure` is that the logs do get rotated over time. For example:
```
[root@dhcp181-3 ~]# ls /var/log/ | grep secure
secure
secure-20191110
secure-20191117
secure-20191124
secure-20191201
```

I think we can change the command to include the logs files that were rotated out of circuit. For example,
```
[root@dhcp181-3 ~]# grep userdel /var/log/secure*
/var/log/secure:Dec  2 10:28:39 dhcp181-3 userdel[26270]: delete user 'tmp_user'
/var/log/secure:Dec  2 10:28:39 dhcp181-3 userdel[26270]: removed group 'tmp_user' owned by 'tmp_user'
/var/log/secure:Dec  2 10:28:39 dhcp181-3 userdel[26270]: removed shadow group 'tmp_user' owned by 'tmp_user'
/var/log/secure-20191201:Nov 27 09:34:21 dhcp181-3 userdel[9776]: delete user 'newuser'
/var/log/secure-20191201:Nov 27 09:34:21 dhcp181-3 userdel[9776]: removed group 'newuser' owned by 'newuser'
/var/log/secure-20191201:Nov 27 09:34:21 dhcp181-3 userdel[9776]: removed shadow group 'newuser' owned by 'newuser'
```

Reply:
Great suggestion! I will make this change 👍